### PR TITLE
new plugin: exit_delete_from_registry

### PR DESCRIPTION
**Pull request by mmilata**

### CODE REVIEW

File: atomic_reactor/plugins/exit_delete_from_registry.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            for tag, digest in push_conf_registry.digests.items():
+                repo = tag.split(':')[0]
+                url = registry + "/v2/" + repo + "/manifests/" + digest
+                insecure = push_conf_registry.insecure
+                response = requests.delete(url, verify=not insecure, auth=auth)
+
+                if response.status_code == 202:
+                    self.log.info("deleted manifest %s/%s@%s", registry_noschema, repo, digest)
+                elif response.status_code == 404:
+                    self.log.warning("cannot delete %s/%s@%s: not found", registry_noschema, repo, digest)
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_delete_from_registry.py]: We should also handle requests.codes.method_not_allowed with a warning not a failure.

File: atomic_reactor/plugins/exit_delete_from_registry.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                else:
+                    msg = "failed to delete %s/%s@%s: %s" % (registry_noschema, repo, digest, r.reason)
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_delete_from_registry.py]: response.reason?

File: atomic_reactor/plugins/exit_delete_from_registry.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                    self.log.error("%s\n%s", msg, r.text)
```
Comment (twaugh) [inline atomic_reactor/plugins/exit_delete_from_registry.py]: response.text?

File: atomic_reactor/plugins/exit_delete_from_registry.py (hunk 2 above)
Comment (mmilata) [inline atomic_reactor/plugins/exit_delete_from_registry.py]: Oops.

File: atomic_reactor/plugins/exit_delete_from_registry.py (hunk 1 above)
Comment (mmilata) [inline atomic_reactor/plugins/exit_delete_from_registry.py]: Added.

File: docs/plugins.md [hunk 4]
```diff
@@ -172,3 +172,6 @@ These are run at the end of the build, even for failed builds.
  * **sendmail**
    * Status: not yet enabled (chain rebuilds)
    * If this build was triggered by a chain in a parent layer, rather than having been explicitly requested by a developer, email is sent to the image owner(s) about the success or failure of the build.
+ * **delete_from_registry**
+   * Status: enable
```
Comment (twaugh) [inline docs/plugins.md]: Oh, right, docs :)
should be 'enabled' I think, not worth holding up the merge for though.

**Comment by twaugh** [comment]
LGTM

File: docs/plugins.md (hunk 4 above)
Comment (mmilata) [inline docs/plugins.md]: Good catch, updated:)

**Comment by lcarva** [comment]
LGTM

**Pull request merged**